lib/light-service/orchestrator.rb

```diff
@@ -68,6 +68,22 @@ def iterate(collection_key, steps)
         end
       end
 
+      def with_callback(action, steps)
+        lambda do |ctx|
+          ctx.reset_skip_remaining! unless ctx.failure?
+          previous_callback = ctx[:callback]
+
+          ctx[:callback] = lambda do |context|
```

Comment:
Update: This is the _critical line of code_ that made me sure of myself that `steps` are the callbacks. With that said, it can be so easy to gloss over once you're used to `steps` being the main `steps`